src/main/kotlin/com/github/avrokotlin/avro4k/SerialDescriptor.kt

```diff
@@ -1,46 +1,25 @@
 package com.github.avrokotlin.avro4k
 
-import kotlinx.serialization.DeserializationStrategy
 import kotlinx.serialization.ExperimentalSerializationApi
-import kotlinx.serialization.KSerializer
-import kotlinx.serialization.SerializationException
-import kotlinx.serialization.descriptors.PolymorphicKind
-import kotlinx.serialization.descriptors.SerialDescriptor
-import kotlinx.serialization.descriptors.SerialKind
-import kotlinx.serialization.descriptors.capturedKClass
-import kotlinx.serialization.descriptors.elementDescriptors
+import kotlinx.serialization.descriptors.*
 import kotlinx.serialization.modules.SerializersModule
-import kotlinx.serialization.modules.SerializersModuleCollector
-import kotlin.reflect.KClass
 
 @ExperimentalSerializationApi
-fun SerialDescriptor.leavesOfSealedClasses() : List<SerialDescriptor> {
-   return if (this.kind == PolymorphicKind.SEALED) {
-      elementDescriptors.filter {it.kind == SerialKind.CONTEXTUAL }.flatMap { it.elementDescriptors }.flatMap { it.leavesOfSealedClasses() }
-   } else {
-      listOf(this)
-   }
+fun SerialDescriptor.leavesOfSealedClasses(): List<SerialDescriptor> {
+    return if (this.kind == PolymorphicKind.SEALED) {
+        elementDescriptors.filter { it.kind == SerialKind.CONTEXTUAL }.flatMap { it.elementDescriptors }
+            .flatMap { it.leavesOfSealedClasses() }
+    } else {
+        listOf(this)
+    }
 }
 
 @ExperimentalSerializationApi
-fun SerialDescriptor.explicitlyNamedSubclassesFrom(serializersModule: SerializersModule): List<SerialDescriptor> {
-   return if (this.kind == PolymorphicKind.OPEN) {
-      val captured = mutableListOf<Pair<KClass<*>, KSerializer<*>>>()
-      val collector = object : SerializersModuleCollector {
-         override fun <T : Any> contextual(kClass: KClass<T>, provider: (typeArgumentsSerializers: List<KSerializer<*>>) -> KSerializer<*>) {
-         }
-
-         override fun <Base : Any, Sub : Base> polymorphic(baseClass: KClass<Base>, actualClass: KClass<Sub>, actualSerializer: KSerializer<Sub>) {
-            captured.add(baseClass to actualSerializer)
-         }
+fun SerialDescriptor.possibleSerializationSubclasses(serializersModule: SerializersModule): List<SerialDescriptor> {
+    return when (this.kind) {
+        PolymorphicKind.SEALED -> this.leavesOfSealedClasses()
```

Comment:
I'm hoping without much work we should be able to get this working for sealed interfaces. I think the fix wouldn't be here but would be in the part of the code that detects the `kind` based on the input. Currently it's treating `sealed interface` as `PolymorphicKind.OPEN` rather than `PolymorphicKind.SEALED` but I'm not exactly sure where that detection is done - do you know how we might do that?

Reply:
I will give the support of sealed interfaces a try in a separate commit as my time resources are limited right now.